framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,718 @@
+      self.raiseADebug('Trajectory {} has converged successfully {} time(s)!'.format(traj, self._convergenceInfo[traj]['persistence']))
+      if self._convergenceInfo[traj]['persistence'] >= self._requiredPersistence:
+        self._closeTrajectory(traj, 'converge', 'converged', optVal)
+    else:
+      self._convergenceInfo[traj]['persistence'] = 0
+      self.raiseADebug('Resetting convergence for trajectory {}.'.format(traj))
+
+  def _addToSolutionExport(self, traj, rlz, acceptable):
+    """
+      Contributes additional entries to the solution export.
+      @ In, traj, int, trajectory which should be written
+      @ In, rlz, dict, collected point
+      @ In, acceptable, bool, acceptability of opt point
+      @ Out, toAdd, dict, additional entries
+    """
+    # meta variables
+    solution = {'Temp': self.T,
+                'fraction': self.getIteration(traj)/self.limit
+                }
+
+    for var in self.toBeSampled:
+      solution['amp_'+var] = self.info['amp_'+var]
+      solution['delta_'+var] = self.info['delta_'+var]
+
+    for var, val in self.constants.items():
+      solution[var] = val
+
+    solution = dict((key, np.atleast_1d(val)) for key, val in solution.items())
+    for key, val in self._convergenceInfo[traj].items():
+      solution['conv_{}'.format(key)] = bool(val)
+    return solution
+
+  def _formatSolutionExportVariableNames(self, acceptable):
+      """
+        Does magic formatting for variables, based on this class's needs.
+        Extend in inheritors as needed.
+        @ In, acceptable, set, set of acceptable entries for solution export for this entity
+        @ Out, new, set, modified set of acceptable variables with all formatting complete
+      """
+      # remaking the list is easier than using the existing one
+      acceptable = RavenSampled._formatSolutionExportVariableNames(self, acceptable)
+      new = []
+      print('DEBUGG acceptable:', acceptable)
+      while acceptable:
+        template = acceptable.pop()
+        if '{CONV}' in template:
+          new.extend([template.format(CONV=conv) for conv in self._convergenceCriteria])
+        elif '{VAR}' in template:
+          new.extend([template.format(VAR=var) for var in self.toBeSampled.keys()])
+        else:
+          new.append(template)
+      return set(new)
+
+  def _rejectOptPoint(self, traj, info, old):
+    """
+      Having rejected the suggested opt point, take actions so we can move forward
+      @ In, traj, int, identifier
+      @ In, info, dict, meta information about the opt point
+      @ In, old, dict, previous optimal point (to resubmit)
+    """
+    self._cancelAssociatedJobs(info['traj'], step=info['step'])
+    ## what do do if a point is rejected?
+    # TODO user option to EITHER rerun opt point OR cut step!
+    # initialize a new step
+    self._initializeStep(traj)
+    #self._acceptRerun[traj] = True #TODO: What if the model is inherintly stochastic ?!
```

Comment:
remove

Reply:
```#self._acceptRerun[traj] = True```

Reply:
But what about the question I had after TODO: